proxy/proxy.go

```diff
@@ -210,7 +211,8 @@ func New(opts *Options) (*Proxy, error) {
 	}
 
 	p := &Proxy{
-		mux: make(map[string]http.Handler),
+		mux:      make(map[string]http.Handler),
```

Comment:
knew we'd eventually need something like a wrapper struct that would contain a handler, policy, and any other (future) options... Something like `RouteConfig`. 

But I think this makes sense for now.

Reply:
I thought about that as well. I was trying to be minimally intrusive, but I think it makes sense to refactor it now.